### PR TITLE
Fix build of GLFence.cpp for !USE(LIBEPOXY)

### PR DESCRIPTION
**Pull request by donny-dont**
#### a99db22709faf65bf67a14ab2073e02fff4b4206
<pre>
Fix build of GLFence.cpp for !USE(LIBEPOXY)
<a href="https://bugs.webkit.org/show_bug.cgi?id=273597">https://bugs.webkit.org/show_bug.cgi?id=273597</a>

Reviewed by Adrian Perez de Castro and Miguel Gomez.

The OpenGL ES functions require version 3.

* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/platform/graphics/egl/GLFence.cpp:

Canonical link: <a href="https://commits.webkit.org/278262@main">https://commits.webkit.org/278262@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/8b3aaf57fb93b82f107fcaf913d85616dfa1686d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49970 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29258 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2252 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53213 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/647 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52271 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35359 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/216 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40756 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52068 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/26845 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42983 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21888 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/24278 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/222 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8340 "Built successfully") | 
| | [loading 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/46346 "Build is in progress. Recent messages:OS: Sonoma (14.4.1), Xcode: 15.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-api-tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/270 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/54794 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25064 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/201 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48147 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26322 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43185 "Passed tests") | [loading-orange 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47197 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; 4 api tests failed or timed out; 6 api tests failed or timed out; Compiled WebKit; Running run-api-tests-without-change") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10971 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27184 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26051 "Built successfully") | | |

### CODE REVIEW

**Review by aperezdc — APPROVED** [review]
This looks fine.

For reference: we still want to maintain the WPE and GTK ports usable (and buildable) with GLESv2, but given they both use Epoxy they won't use the `GLES3/gl3.h` header 👌🏼

**Review by magomez — APPROVED** [review]
This indeed fixes the build, despite it's not the perfect fix. It's true that the current implementation of GLFence is using only the gles3 version of the functions, and gles3 would be needed for that, but it's also true that gles2 would be enough if the GL_APPLE_sync extension was present. The problem with the latter is that the GLFence is missing the implementation using the extension defines and functions.

In summary, if you don't mind depending on gles3, this is fine. If you want to also support gles2 with the GL_APPLE_sync extension, then you need to add the code for that.

**Comment by rkirsling** [comment]
I'm super confused. This caused the PlayStation build to go _from_ green _to_ red.
